packages/runtime/datastore/src/localChannelContext.ts

```diff
@@ -84,23 +108,95 @@ export class LocalChannelContext implements IChannelContext {
     }
 
     public getAttachSnapshot(): ITree {
+        assert(this.isLoaded && this.channel !== undefined, "Channel should be loaded to take snapshot");
         return snapshotChannel(this.channel);
     }
 
+    private async loadChannel(): Promise<IChannel> {
+        assert(!this.isLoaded, "Channel must not already be loaded when loading");
+        assert(this.snapshotTree, "Snapshot should be provided to load from!!");
+
+        assert(await this.services.objectStorage.contains(".attributes"), ".attributes blob should be present");
+        const attributes = await readAndParse<IChannelAttributes>(
+            this.services.objectStorage,
+            ".attributes");
+
+        assert(this.factory, "Factory should be there for local channel");
+        const channel = await this.factory.load(
+            this.runtime,
+            this.id,
+            this.services,
+            undefined,
+            attributes);
+
+        // Commit changes.
+        this.channel = channel;
+        this._isLoaded = true;
+
+        if (this.attached) {
+            this.channel.connect(this.services);
+        }
+
+        // Send all pending messages to the channel
+        for (const message of this.pending) {
+            try {
+                this.services.deltaConnection.process(message, false, undefined /* localOpMetadata */);
+            } catch (err) {
+                // record sequence number for easier debugging
+                const error = CreateContainerError(err);
+                error.sequenceNumber = message.sequenceNumber;
+                throw error;
+            }
+        }
+        return this.channel;
+    }
+
     public attach(): void {
         if (this.attached) {
             throw new Error("Channel is already attached");
         }
 
-        const services = createServiceEndpoints(
-            this.channel.id,
-            this.dataStoreContext.connected,
-            this.submitFn,
-            this.dirtyFn,
-            this.storageService);
-        this.connection = services.deltaConnection;
-        this.channel.connect(services);
-
+        if (this.isLoaded) {
+            assert(this.channel, "Channel should be there if loaded!!");
+            this.channel.connect(this.services);
+        }
         this.attached = true;
     }
+
+    private get services() {
```

Comment:
So we were depending on previous version of common utils package where the lazy class is not exported. So i have exposed that here and then will follow up with another Pr to use that class here, otherwise this Pr will become huge.